src/dfeed/web/web/view/group.d

```diff
@@ -169,6 +171,130 @@ void discussionGroup(GroupInfo groupInfo, int page)
+				html.put(`title="`); html.putEncodedEntities(thread.firstPost.time.formatTimeLoc!"l, d F Y, H:i:s e"); html.put(`">`);
+				html.put(`<span class="short">`); html.putEncodedEntities(formatTinyTime(thread.firstPost.time)); html.put(`</span>`);
+				html.put(`<span class="long">`); html.putEncodedEntities(formatShortTime(thread.firstPost.time, false)); html.put(`</span>`);
+			html.put(`</time>`);
+			string threadGravatarHash = getGravatarHash(thread.firstPost.authorEmail);
+			html.put(`<img  class="firstpost-author-image" src="//www.gravatar.com/avatar/`, threadGravatarHash, `?d=identicon">`);
+			html.put(`<div class="firstpost-author-name">`);
+					html.putEncodedEntities(thread.firstPost.author);
+			html.put(`</div>`);
+			html.put(`<div class="replies-total">`);
+				html.put(`<span class="short">`);
+					html.put(formatNumber(thread.postCount - 1));
+				html.put(`</span>`);
+				html.put(`<span class="long">`);
+					html.put(formatNumber(thread.postCount - 1), " ", (thread.postCount - 1) == 1 ? _!"reply" : _!"replies");
+				html.put(`</span>`);
+			html.put(`</div>`);
+			html.put(`<div class="replies-new">`);
+				html.put(`<span class="short">`);
+					if (thread.unreadPostCount && thread.unreadPostCount != thread.postCount)
+						html.put(`(<a href="`, idToUrl(thread.id, "first-unread"), `">`, formatNumber(thread.unreadPostCount), `</a>)`);
+				html.put(`</span>`);
+				html.put(`<span class="long">`);
+					if (thread.unreadPostCount && thread.unreadPostCount != thread.postCount)
+						html.put(`<a href="`, idToUrl(thread.id, "first-unread"), `">`, formatNumber(thread.unreadPostCount), ` `, _!`new`, `</a>`);
+				html.put(`</span>`);
+			html.put(`</div>`);
+			html.put(`<div class="lastpost-time">`);
+				html.put(`<a `);
+					html.put(`href="`); html.putEncodedEntities(idToUrl(thread.lastPost.id)); html.put(`">`);
+					html.put(_!`Last Post`);
+					html.put(`: `);
+					html.put(`<time `);
+						html.put(`datetime="`); html.putEncodedEntities(thread.lastPost.time.formatTimeLoc!"c"); html.put(`" `);
+						html.put(`title="`); html.putEncodedEntities(thread.lastPost.time.formatTimeLoc!"l, d F Y, H:i:s e"); html.put(`">`);
+						html.put(`<span class="short">`); html.putEncodedEntities(formatTinyTime(thread.lastPost.time)); html.put(`</span>`);
+						html.put(`<span class="long">`); html.putEncodedEntities(formatShortTime(thread.lastPost.time, false)); html.put(`</span>`);
+					html.put(`</time>`);
+				html.put(`</a>`);
+			html.put(`</div>`);
+			string lastpostGravatarHash = getGravatarHash(thread.lastPost.authorEmail);
+			html.put(`<div class="lastpost-author">`);
+				html.put(`<img class="lastpost-author-image" src="//www.gravatar.com/avatar/`, lastpostGravatarHash, `?d=identicon">`);
+				html.put(`<span class="lastpost-author-name">`);
+					html.putEncodedEntities(thread.lastPost.author);
+				html.put(`</span>`);
+			html.put(`</div>`);
+		html.put(`</div>`);
```

Comment:
The multiple `html.put` calls will be worse for performance than aggregating them, because the appender tries to perform up to one allocation per call. This is also going to be really annoying to edit for anyone with an editor which auto-formats code.

In general, when you contribute to any existing project, you are going to be asked to keep the existing code style, and in this case it's no exception.